src/main_no_dnn.py

```diff
@@ -0,0 +1,168 @@
+        )
+
+        outputs = self.model(**inputs, position_ids=position_ids)
+
+        return outputs.pooler_output
+
+
+class Trainer:
+    def __init__(self, config):
+        self.config = config
+
+        self.dataset_train = TextClassificationDataset(config, "train")
+        self.dataset_eval = TextClassificationDataset(config, "eval")
+
+        self.vectorizer = self.__create_vectorizer(config.vectorizer_type)
+
+    def __create_vectorizer(self, vectorizer_type: str) -> FeatureExtractorBase:
+        if args.vectorizer_type == "tfidf":
+            return FeatureExtractorTfidf(self.config)
+        if args.vectorizer_type == "use":
+            return FeatureExtractorUse(self.config)
+        if args.vectorizer_type == "roberta":
+            return FeatureExtractorRoberta(self.config)
+
+    def __create_models(self, model_type: list[str]):
+        n_ens = 100
+        return [
+            (dummy.DummyClassifier(strategy="stratified"), "dummy"),
+            (tree.DecisionTreeClassifier(), "decision tree"),
+            (ensemble.RandomForestClassifier(n_estimators=n_ens), "random forest"),
+            (ensemble.ExtraTreesClassifier(n_estimators=n_ens), "extra tree"),
+            (
+                lgb.LGBMClassifier(
+                    objective="multiclass", num_class=self.dataset_train.n_labels
+                ),
+                "lightgbm",
+            ),
+            (svm.SVC(), "svc"),
+            (neighbors.KNeighborsClassifier(), "knn"),
+        ]
+
+    def __run_model(self, model, name, X_train, X_eval, y_train, y_eval):
+        start = time.time()
+        model.fit(X_train, y_train)
+        print(f"[{name}] {time.time() - start}")
+        y_pred = model.predict(X_eval)
+        print(metrics.classification_report(y_eval, y_pred))
+
+    def train(self):
+        vectors_train = self.vectorizer.vectorize(self.dataset_train.texts)
+        vectors_eval = self.vectorizer.vectorize(self.dataset_eval.texts)
+
+        for (model, name) in self.__create_models([]):
+            if self.config.balanced:
+                model = BalancedBaggingClassifier(base_estimator=model)
+
+            self.__run_model(
+                model,
+                name,
+                vectors_train,
+                vectors_eval,
+                self.dataset_train.labels.argmax(axis=1),
+                self.dataset_eval.labels.argmax(axis=1),
+            )
+
+
+@dataclass
+class Config:
+    vectorizer_type: str = field(
+        default="use", metadata=dict(type=str, choices=["tfidf", "use", "roberta"])
+    )
+    balanced: bool = field(default=False, metadata=dict(type=bool))
+    dataroot: str = field(default="data/debug", metadata=dict(type=str))
+    data_file: str = field(default=None, metadata=dict(type=str))
+    label_file: str = field(default=None, metadata=dict(type=str))
+    logger: Logger = field(default=getLogger(__name__), metadata=dict(type=Logger))
```

Comment:
need to set metadata.type to use argparse-dataclass with __future__.annotations. this is because field.type of dataclass instance is string instead of type.